src/MainWindow.vala

```diff
@@ -271,6 +274,46 @@ public class Tasks.MainWindow : Hdy.ApplicationWindow {
         });
     }
 
+    private void dialog_refresh_collection_error (Error e) {
+        string error_message = e.message;
+
+        GLib.Idle.add (() => {
+            var error_dialog = new Granite.MessageDialog (
+                _("Refresh collection failed"),
+                _("The backend may be unavailable or unable to read from."),
+                new ThemedIcon ("dialog-error"),
+                Gtk.ButtonsType.CLOSE
+            ) {
+                transient_for = this
+            };
+            error_dialog.show_error_details (error_message);
+            error_dialog.run ();
+            error_dialog.destroy ();
+
+            return GLib.Source.REMOVE;
+        });
+    }
+
+    private void dialog_refresh_task_list_error (Error e) {
+        string error_message = e.message;
+
+        GLib.Idle.add (() => {
+            var error_dialog = new Granite.MessageDialog (
+                _("Refresh task list failed"),
+                _("The backend may be unavailable or unable to read from."),
```

Comment:
We should probably try to be as accurate and helpful here as we have the information to be. It would be nice to use more natural sounding language as well. Something like:
```
Unable to get updates for "Groceries"
The tasks at "me@example.com" may be unavailable or unable to be read."
```

I wonder about that second line as well. If we didn't fail to refresh the collection, then is it really a possibility that the _backend_ is unavailable? Should this instead say that this task list may be unavailable?

Reply:
Changed the error message accordingly. Also, you're right - the backend is indeed available in this case so I changed the message to reflect this.